tests/integration/docker/dns.bats

```diff
@@ -0,0 +1,86 @@
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# DNS tests using Swarm
+
+SRC="${BATS_TEST_DIRNAME}/../../lib/"
+# number of replicas that will be launch
+number_of_replicas=1
+# total number of replicas launch with nginx
+total_replicas=2
+# number of attempts
+number_of_attempts=5
+
+#This function will verify that swarm is not running or it finishes
+function clean_swarm_status() {
+	for j in `seq 0 $number_of_attempts`; do
+		if $DOCKER_EXE node ls; then
+			$DOCKER_EXE swarm leave --force
+			# docker swarm leave is not inmediately so it requires time to finish
+			sleep 5
+		else
+			break
+		fi
+	done
+}
+
+setup() {
+	source $SRC/test-common.bash
+	runtime_docker
+	clean_swarm_status
+	interfaces=($(readlink /sys/class/net/* | grep -i pci | xargs basename -a))
+	swarm_interface_arg=""
+	for i in ${interfaces[@]}; do
+		if [ "`cat /sys/class/net/${i}/operstate`" == "up" ]; then
+			swarm_interface_arg="--advertise-addr ${i}"
+			break;
+		fi
+	done
+	$DOCKER_EXE swarm init ${swarm_interface_arg}
+	$DOCKER_EXE service create --name testswarm1 --replicas $number_of_replicas --publish 8080:80 nginx /bin/bash -c "hostname > /usr/share/nginx/html/hostname; nginx -g \"daemon off;\""
+	$DOCKER_EXE service create --name testswarm2 --replicas $number_of_replicas --publish 8082:80 nginx /bin/bash -c "hostname > /usr/share/nginx/html/hostname; nginx -g \"daemon off;\""	
+	$DOCKER_EXE service create --name testdns --replicas $number_of_replicas --publish 8084:80 mcastelino/nettools sleep 60000
+	while [ `$DOCKER_EXE ps --filter status=running --filter ancestor=nginx:latest -q | wc -l` -lt $total_replicas ]; do
+		sleep 1
+	done
+	while [ `$DOCKER_EXE ps --filter status=running --filter ancestor=mcastelino/nettools:latest -q | wc -l` -lt $number_of_replicas ]; do
+		sleep 1
+	done
+}
+
+@test "DNS test to check Server and IP Address" {
+	skip "this is not working (this is related with https://github.com/01org/cc-oci-runtime/issues/578)"
+	container_id=`$DOCKER_EXE ps -qaf "name=testdns"`
+	$DOCKER_EXE exec -it $container_id nslookup testswarm1
+	$DOCKER_EXE exec -it $container_id nslookup testswarm2	
+}
+
+@test "DNS test to check hostname" {
+	skip "this is not working (this is related with https://github.com/01org/cc-oci-runtime/issues/578)"
+	container_id=`$DOCKER_EXE ps -qaf "name=testdns"`
+	$DOCKER_EXE exec -it $container_id nslookup curl testswarm1:/hostname
```

Comment:
Typo. This should just be $DOCKER_EXE exec -it $container_id curl testswarm1:/hostname
As this ensures we can reach the ngnix service via the dns name